### PR TITLE
Add a KeepAlive on connection between layer and agent

### LINKED ISSUE #63: After about 10 minutes of idle time, mirrord-layer starts reading empty messages
### Bug Description

This isn't the result of the agent terminating, as logs indicate it's still alive when that happens.

### Steps to Reproduce

Start a process wrapped by mirrord-layer and wait for about 10 minutes.

### Backtrace

_No response_

### Your operating system and version

MacOS

### Your Node version (`node --version`)

17.6.0

### Additional Info

_No response_

### PR DESCRIPTION
**Pull request by infiniteregrets**
Closes #63. The layer sends a ping to the agent pod and the pod responds back with a pong.

### CODE REVIEW

**Comment by infiniteregrets** [comment]
Todo: currently ping works just fine as seen in the logs, but pong is never received.

File: mirrord-layer/src/lib.rs [hunk 1]
```diff
@@ -143,6 +144,11 @@ async fn handle_hook_message(
                 });
         }
     }
+    loop {
```
Comment (aviramha) [inline mirrord-layer/src/lib.rs:147]: This needs to be in the timeout section of the select, also we need that when it occurs we check if last ping was pongd or not

File: mirrord-protocol/src/codec.rs [hunk 2]
```diff
@@ -31,11 +31,17 @@ pub struct LogMessage {
     pub message: String,
 }
 
+#[derive(Encode, Decode, Debug, PartialEq, Eq, Clone)]
+pub struct PingMessage {
```
Comment (aviramha) [inline mirrord-protocol/src/codec.rs:35]: redundant

**Review by aviramha — CHANGES_REQUESTED** [review]

**Review by eyalb181 — CHANGES_REQUESTED** [review]
LGTM, just update the CHANGELOG.

**Review by eyalb181 — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -9,6 +9,8 @@ Check [Keep a Changelog](http://keepachangelog.com/) for recommendations on how
 ### Added
 - Prompt user to update if their version is outdated in the VS Code extension or CLI.
 
+- Added a keep-alive to keep the agent-pod from exiting, closes [#63](https://github.com/metalbear-co/mirrord/issues/63)
```
[the remark this answers is not in the document]
Comment (infiniteregrets) [reply CHANGELOG.md]: done, thanks

**Review by eyalb181 — APPROVED** [review]

**Review by eyalb181 — APPROVED** [review]

**Review by aviramha — APPROVED** [review]

**Pull request merged**